### PR TITLE
allow double-digit versions in XcodeVersion

### LINKED ISSUE #1849: Could not get CLTVersion
if CLT version is over 10.*, it could not get the version.
`version = re.match(r'(\d+\.\d+\.?\d*)', version).groups()`

and now latest CLT version is 10.14.

https://developer.apple.com/download/more/?=command%20line%20tools

https://github.com/nodejs/node-gyp/blob/872ee89a8541264ee2d96e162d410f478bba7815/gyp/pylib/gyp/xcode_emulation.py

### PR DESCRIPTION
**Pull request by roryqueue**
Fixes #1849

##### Checklist

- [ ] `npm install && npm test` passes
- [ ] tests are included 
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

**Comment by cclauss** [comment]
Related to #1849 ?

**Comment by roryqueue** [comment]
yes, relates-- sorry at work and not able to dig further atm, but that change seems to fix

**Comment by roryqueue** [comment]
(or rather, obviously doesn't work given it's failing, but the regex change matches both 9.X and 10.X, where as the other one fails on 10.X, so could be a hint to a real fix)

**Comment by rvagg** [comment]
Sorry, I'm not a python person but this looks invalid to me, `\d+` will match double digits just as well as `\d?\d` in any normal regex engine. Can I close this or is someone willing to dispute that Python does something funky with regexes that makes this necessary?

**Comment by cclauss** [comment]
Can we just replace this line with __".".join(version.split(".")[:3]__ and move on?  Not a fan of regex for trivial use cases.

File: gyp/pylib/gyp/xcode_emulation.py [hunk 1]
```diff
@@ -1277,7 +1277,7 @@ def XcodeVersion():
   except:
     version = CLTVersion()
     if version:
-      version = re.match(r'(\d+\.\d+\.?\d*)', version).groups()[0]
+      version = re.match(r'(\d?\d\.\d?\d\.?\d?\d*)', version).groups()[0]
     else:
       raise GypError("No Xcode or CLT version detected!")
```
Comment (cclauss) [inline gyp/pylib/gyp/xcode_emulation.py]: On exception, let's add the version that was found...
` raise GypError("No Xcode or CLT version detected!  {}".format(version)`

**Comment by cclauss** [comment]
I believe that #1890 fixed this. Please rebase and reverify.

**Comment by cclauss** [comment]
Conflicts to resolve.

**Comment by rvagg** [comment]
I think this is dealt with already and the solution posed here isn't correct as far as I can tell

**Pull request closed without merging**